reward/supply_manager.go

```diff
@@ -379,9 +380,35 @@ func (sm *supplyManager) totalSupplyFromState(num uint64) (*big.Int, error) {
 	return totalSupply, nil
 }
 
+func (sm *supplyManager) getAccRewardUncached(num uint64) (*database.AccReward, error) {
+	accReward := sm.db.ReadAccReward(num)
+	if accReward != nil {
+		return accReward, nil
+	}
+
+	// Trace back to the last stored accumulated reward.
+	var fromNum uint64
+	var fromAcc *database.AccReward
+	for i := uint64(1); i < supplyReaccLimit; i++ {
+		accReward = sm.db.ReadAccReward(num - i)
+		if accReward != nil {
+			fromNum = num - i
+			fromAcc = accReward
+			break
+		}
+	}
+	if fromAcc == nil {
+		return nil, errNoAccReward
+	}
+
+	logger.Trace("on-demand reaccumulating rewards", "from", fromNum, "to", num)
+	return sm.accumulateReward(fromNum, num, fromAcc, false)
```

Comment:
I think `write` is meaningless for this since in most cases there's no checkpoint interval between `fromNum` and `num`. And even if there's a case, I think writing missing `accReward` at checkpoint to db isn't needed to be prevented.

Reply:
To prevent overwriting Last number, and to keep database size from growing via API.

Reply:
Oh, sure. I got the point.